src/firstpass.rs

```diff
@@ -1490,6 +1638,56 @@ where
     (ix, None)
 }
 
+/// Parses an attribute block content, such as `.class1 #id .class2`.
+///
+/// Returns `(id, classes)`.
+///
+/// It is callers' responsibility to find opening and closing characters of the attribute
+/// block. Usually [`extract_attribute_block_from_header_text`] method does it for you.
+///
+/// Note that this parsing requires explicit whitespace separators between
+/// attributes. This is intentional design with the reasons below:
+///
+/// * to keep conversion simple and easy to understand for any possible input,
+/// * to avoid adding less obvious conversion rule that can reduce compatibility
+///   with other implementations more, and
+/// * to follow the major design of implementations with the support for the
+///   attribute blocks extension (as of this writing).
+///
+/// See also: [`Options::ENABLE_HEADING_ATTRIBUTES`].
+///
+/// [`Options::ENABLE_HEADING_ATTRIBUTES`]: `crate::Options::ENABLE_HEADING_ATTRIBUTES`
+/// [`extract_attribute_block_from_header_text`]:
+///     `FirstPass::extract_attribute_block_from_header_text`
+fn parse_inside_attribute_block(inside_attr_block: &str) -> (Option<&str>, Vec<&str>) {
+    let mut id = None;
+    let mut classes = Vec::new();
+
+    for attr in inside_attr_block.split_ascii_whitespace() {
```

Comment:
This method is good, but I just want to let you know that [`split_at`](https://doc.rust-lang.org/std/primitive.slice.html#method.split_at) exists. With it, one could write the bit inside loop as this:
```rust
match attr.as_bytes().split_at(1) {
    (_, []) => {},
    ([b'#'], tail) => { id = Some(tail) },
    ([b'.'], tail) => { classes.push(tail) },
    _ => {},
}
```

Reply:
Thank you!
I'll use `str::split_at` since the function wants ID fragment and class names to be `&str`.
[The implementation of `str::split_at`](https://doc.rust-lang.org/1.56.0/src/core/str/mod.rs.html#570) seems to be O(1) and efficient enough.